### PR TITLE
Add an operation to rename a file after it has been exported

### PR DESCRIPTION
**Pull request by guewen**
- Add fields on tasks to allow to rename files after they have been moved
- Add a context manager to allow to share the same connection for several operations on a ftp/sftp/...

### CODE REVIEW

**Comment by bealdav** [comment]
ping @bguillot @sebastienbeau for review, thanks

File: file_repository/file_connexion.py [hunk 1]
```diff
@@ -40,19 +42,17 @@ def open_and_close_connection(func):
...
             self.connect()
-            try:
-                response = func(self, *args, **kwargs)
-            except:
-                raise
-            finally:
-                self.close()
-            return response
+
+        # either the connection is persistent,
```
Comment (bealdav) [inline file_repository/file_connexion.py:48]: persistant

persistant

**Comment by bealdav** [comment]
seems ok for me. ping @Viggor thanks to share you opinion

**Comment by Viggor** [comment]
Seems ok for me too, i will use your rename function and logical in the future module which will replace file-exchange.

File: file_repository/file_connexion.py (hunk 1 above)
Comment (guewen) [inline file_repository/file_connexion.py:48]: I persist, it spells "persistent" in english :smiling_imp:

File: file_repository/file_connexion.py [hunk 2]
```diff
@@ -40,19 +42,17 @@ def open_and_close_connection(func):
     """
     @functools.wraps(func)
     def wrapper(self, *args, **kwargs):
-        if self.persistant:
-            if not self.connection:
-                self.connect()
-            return func(self, *args, **kwargs)
-        else:
+        if self.persistant and not self.connection:
```
Comment (bealdav) [inline file_repository/file_connexion.py:45]: persistent ;-)

Comment (guewen) [inline file_repository/file_connexion.py:45]: ummm, do you want me to correct this typo? It was there already and this it is a public attribute, I'm not sure it is safe to rename it.

**Comment by guewen** [comment]
Any news on this PR?

**Comment by guewen** [comment]
Thanks for the merge!

**Pull request merged**